plugins/node/opentelemetry-instrumentation-pg/src/instrumentation.ts

```diff
@@ -173,101 +173,136 @@ export class PgInstrumentation extends InstrumentationBase {
+                text: utils.addSqlCommenterComment(
+                  span,
+                  (arg0 as utils.ObjectWithText).text
+                ),
+              }
+            : args[0];
         }
 
-        // Bind callback to parent span
+        // Bind callback (if any) to parent span (if any)
         if (args.length > 0) {
           const parentSpan = trace.getSpan(context.active());
           if (typeof args[args.length - 1] === 'function') {
             // Patch ParameterQuery callback
             args[args.length - 1] = utils.patchCallback(
-              plugin.getConfig(),
+              instrumentationConfig,
               span,
-              args[args.length - 1] as PostgresCallback
+              args[args.length - 1] as PostgresCallback // nb: not type safe.
             );
+
             // If a parent span exists, bind the callback
             if (parentSpan) {
               args[args.length - 1] = context.bind(
                 context.active(),
                 args[args.length - 1]
               );
             }
-          } else if (
-            typeof (args[0] as NormalizedQueryConfig).callback === 'function'
-          ) {
+          } else if (typeof queryConfig?.callback === 'function') {
             // Patch ConfigQuery callback
             let callback = utils.patchCallback(
               plugin.getConfig(),
               span,
-              (args[0] as NormalizedQueryConfig).callback!
+              queryConfig.callback as PostgresCallback // nb: not type safe.
             );
+
             // If a parent span existed, bind the callback
             if (parentSpan) {
               callback = context.bind(context.active(), callback);
             }
 
-            // Copy the callback instead of writing to args.callback so that we don't modify user's
-            // original callback reference
+            // Copy the callback instead of writing to args.callback so that we
+            // don't modify user's original callback reference
             args[0] = { ...(args[0] as object), callback };
           }
         }
 
-        // Perform the original query
-        const result: unknown = original.apply(this, args as any);
+        if (
+          typeof instrumentationConfig.queryHook === 'function' &&
+          queryConfig
+        ) {
+          safeExecuteInTheMiddle(
+            () => {
+              instrumentationConfig.queryHook!(span, {
+                connection: this.connectionParameters,
+                query: {
+                  text: queryConfig.text,
+                  // nb: the casts below are not type safe; they'll be wrong in
+                  // the presence of illegal arguments. But we leave it up to
+                  // the queryHook to handle that, and we catch and swallow any
+                  // errors it throws.
+                  values: queryConfig.values as unknown[],
+                  name: queryConfig.name as string | undefined,
+                },
```

Comment:
nit: The type of this object in `types.ts` is 
```js
query: pgTypes.QueryConfig;
```

Which resolves to 
```js
export interface QueryConfig<I extends any[] = any[]> {
    name?: string | undefined;
    text: string;
    values?: I | undefined;
    types?: CustomTypesConfig | undefined;
}
```

It also includes `types` attribute which I don't know what it is used for, but the instrumentation will not pass it to the hook, where a user might expect that it will.

Reply:
Makes sense. I replaced this with a distinct type that explicitly reflects what's passed to the hook.